### DISCUSSION TITLE
Next Link is giving hydration error

### DISCUSSION THREAD

**Discussion by aly8i** [vercel/community#995] [Help] [answered]
this is the code that is throwing error:
`<li className={styles.li} onClick={()=>handleFilter("")}>
                <Link href="/" passHref>
                  <span>
                    <div className={styles.title}>Home</div>
                  </span>
                  <HomeIcon className={styles.icon}/>
                </Link>
              </li>`
when I remove or comment the link wrapper it works fine. however, when I add it throws hydration error:

> Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

See more info here: https://nextjs.org/docs/messages/react-hydration-error

**Comment by amyegan** [comment, marked as answer]
Hi @aly8i! I believe you need to add the `legacyBehavior` prop to enable legacy nested child behavior for [`next/link`](https://nextjs.org/blog/next-13#nextlink) in v13. There are more example scenarios and details in the [`next/link` documentation](https://nextjs.org/docs/api-reference/next/link). You should check out the list of [breaking changes](https://nextjs.org/blog/next-13#breaking-changes) for an overview of what changes to make when upgrading from Next.js 12 to 13.

**Reply by aly8i** [reply]
@amyegan the hydration error is gone once i added `legacyBehavior`, but the link isn't passed to my browser anymore.

**Reply by amyegan** [reply]
Can you show me what you mean with code, images, or a quick video recording of your screen?

**Reply by aly8i** [reply]
@amyegan I already provided you with the code. a `<ul> <li>` to display a list of icons like a home icon for example. This icon (home) is wrapped with `<link>` so when I click on it, it will take me to the base url. I was using `<Link href="/" passHref><> <div className={styles.title}>Home</div> <HomeIcon className={styles.icon}/> </></Link>` that is giving a hydration error. I added `legacyBeehavior` like so `<Link href="/" passHref legacyBehavior><><div className={styles.title}>Home</div> <HomeIcon className={styles.icon}/> </></Link>` the hydration error is gone. However, the link isn't working anymore. it is not redirecting me to the desired url. So, I removed the next/link and used next/router instead. It worked fine. I couldn't figure out the problem. I felt like you must be informed to fix this issue. The same code used to work fine before the update.

**Reply by amyegan** [reply]
Thanks for the additional explanation. I think the problem might be that you're using a `div` with `legacyBehavior` but not returning an an `a` tag. Based on the documentation, this doesn't seem to be a bug. Does it work if you make the child of your `Link` an `a`?

You should take a look at the [`next/link` documentation](https://nextjs.org/docs/api-reference/next/link) for examples of how to configure `next/link`. However, you can keep using `next/router` if that works better for your case!
